mm2src/coins/solana/solana_coin.rs

```diff
@@ -228,11 +327,84 @@ impl MmCoin for SolanaCoin {
     }
 
     fn spawner(&self) -> WeakSpawner {
-        todo!()
+        self.abortable_system.weak_spawner()
     }
 
     fn withdraw(&self, req: WithdrawRequest) -> WithdrawFut {
-        todo!()
+        let coin = self.clone();
+        let fut = async move {
+            let to = SolanaAddress::from_str(&req.to).map_err(|e| WithdrawError::InvalidAddress(e.to_string()))?;
+
+            let rpc = coin
+                .rpc_client()
+                .await
+                .map_err(|e| WithdrawError::Transport(e.into_inner()))?;
+
+            let lamports = coin.calculate_withdraw_amount(&req).await?;
+
+            if lamports == 0 {
+                return MmError::err(WithdrawError::AmountTooLow {
+                    amount: req.amount,
+                    threshold: coin.min_tx_amount(),
+                });
+            }
+
+            let recent_blockhash = rpc
+                .get_latest_blockhash()
+                .await
+                .map_err(|e| WithdrawError::Transport(e.to_string()))?;
+
+            // Actual TX
+            let tx = solana_system_transaction::transfer(&coin.keypair, &to, lamports, recent_blockhash);
+
+            let tx_hash = tx
+                .signatures
+                .first()
+                .map(|s| s.to_string())
+                .ok_or_else(|| WithdrawError::InternalError("Couldn't find the TX signature.".to_owned()))?;
+
+            let tx_bytes =
+                bincode::serialize(&tx).map_err(|e| MmError::new(WithdrawError::InternalError(e.to_string())))?;
+
+            let tx_data = TransactionData::new_signed(BytesJson(tx_bytes), tx_hash.clone());
+
+            let amount_dec = u64_lamports_to_big_decimal(lamports, SOLANA_DECIMALS);
+
+            let fee = rpc
+                .get_fee_for_message(tx.message())
```

Comment:
Wouldn't it be better to return fee from `calculate_withdraw_amount` to avoid additional calls?